assets/scripts/fe/src/app/components/payroll/payroll-list/payroll-list.component.ts

```diff
@@ -29,4 +31,36 @@ export class PayrollListComponent implements OnInit {
     }
   }
 
+  onToggle(item){
+    // Function trigger upon checkbox toggle
+
+    // Checks if item is already exist in mapDownloadPDF 
+    if(this.payrollservice.mapDownloadPDF.has(item.id)){
+      // if item is exist. Remove it from mapped data
+      this.payrollservice.mapDownloadPDF.delete(item.id);
+    }
+    else{
+      // else, add the item to mapped data
+      this.payrollservice.mapDownloadPDF.set(item.id, item)
+    } 
+  }
+
+  downloadPDF(){
+    // loops over each item of the mapped data
+    this.payrollservice.mapDownloadPDF.forEach((value: boolean, key: string) => {
+      // get file name
+      const file_name = this.getFileName(value);
+      // call the download functionality with parameters of item id and file name
+      this.payrollservice.downloadPDF(key, file_name);
+    });
+  }
+
+  getFileName(payrollValue){
```

Comment:
change this method to return a string in this format.

`P{ user_id }_{ last_name }_{month}-{day}-{year}.pdf`

eg.
`p1_gemenez_4-29-19.pdf`